common/reflect_resource.go

```diff
@@ -575,6 +597,35 @@ func readReflectValueFromData(path []string, d attributeGetter,
 		}
 		return nil
 	})
+	if err != nil {
+		return err
+	}
+	return setForceSendFields(rv, forceSendFields)
+}
+
+func setForceSendFields(rv reflect.Value, presentFields []string) error {
+	if len(presentFields) == 0 {
+		return nil
+	}
+
+	if rv.Kind() != reflect.Ptr && rv.Kind() != reflect.Struct {
+		return nil
+	}
+
+	field := rv.FieldByName("ForceSendFields")
+
+	if !field.IsValid() {
+		return nil
```

Comment:
This is for compatibility with non-SDK structs?

Reply:
That, and also for any struct which does not have the ForceSendFields field. (SDK or not). In such case, `FieldByName` returns the zero(default) Value object, which is not "valid".